### ISSUE TITLE
no pad_token or eos_token in wandb eval table "Eval - Predictions vs Ground Truth"

### ISSUE THREAD

**Issue by BaiMoHan** [axolotl-ai-cloud/axolotl#2330]
### Please check that this issue hasn't been reported before.

- [x] I searched previous [Bug Reports](https://github.com/axolotl-ai-cloud/axolotl/labels/bug) didn't find any similar reports.

### Expected Behavior

sft llama3
should see <eot_id> etc

### Current behaviour

set eval_table_size 1
![Image](https://github.com/user-attachments/assets/e952b0eb-66d3-4f38-9f7d-36fb0b3cc7c4)

### Steps to reproduce

axolotl train xx.yaml

### Config yaml

```yaml
base_model: NousResearch/Meta-Llama-3-8B
model_type: LlamaForCausalLM
tokenizer_type: AutoTokenizer
load_in_8bit: true
load_in_4bit: false
strict: false
chat_template: jinja
chat_template_jinja: "{% if not add_generation_prompt is defined %}{% set add_generation_prompt = false %}{% endif %}{% set loop_messages = messages %}{% for message in loop_messages %}{% set content = '<|start_header_id|>' + message['role'] + '<|end_header_id|>\n\n' + (message['name'] + ': ' if message['role'] != 'system' else '') + message['content'] | trim + '<|eot_id|>' %}{% if loop.index0 == 0 %}{% set content = bos_token + content %}{% endif %}{{ content }}{% endfor %}{% if add_generation_prompt %}{{ '<|start_header_id|>assistant<|end_header_id|>\n\n' }}{% endif %}"
datasets:
  - path: /workspace/axolotl/data.jsonl
    type: chat_template
    field_messages: conversations
    message_field_role: from
    message_field_content: value
dataset_prepared_path:
val_set_size: 0.05
output_dir: ./outputs/lora-out-11
sequence_len: 4096
sample_packing: true
eval_sample_packing: false
pad_to_sequence_len: true
adapter: lora
lora_model_dir:
lora_r: 32
lora_alpha: 16
lora_dropout: 0.05
lora_target_linear: true
lora_fan_in_fan_out:
lora_modules_to_save:
  - embed_tokens
  - lm_head
gradient_accumulation_steps: 4
micro_batch_size: 2
num_epochs: 8
optimizer: adamw_bnb_8bit
lr_scheduler: cosine
learning_rate: 0.0002
train_on_inputs: false
group_by_length: false
bf16: auto
fp16:
tf32: false
gradient_checkpointing: true
early_stopping_patience:
resume_from_checkpoint:
local_rank:
logging_steps: 1
xformers_attention:
flash_attention: true
s2_attention:
warmup_steps: 10
evals_per_epoch: 4
eval_table_size: 1
eval_max_new_tokens: 64
saves_per_epoch: 1
save_total_limit: 12
debug:
deepspeed:
weight_decay: 0.0
fsdp:
fsdp_config:
special_tokens:
   pad_token: <|end_of_text|>
```

### Which Operating Systems are you using?

- [x] Linux
- [ ] macOS
- [ ] Windows

### Python Version

3.10

### axolotl branch-commit

860609392184cf62a7e0ca676658b170e059ce6c

### Acknowledgements

- [x] My issue title is concise, descriptive, and in title casing.
- [x] I have searched the existing issues to make sure this bug has not been reported yet.
- [x] I am using the latest version of axolotl.
- [x] I have provided enough information for the maintainers to reproduce and diagnose the issue.

**Comment by NanoCode012** [comment]
To add more info from discord discussion, the problem stems from the eval_table code, which was written quite some time ago and hasn't been actively maintained. At this point, I'm not certain if there's an interest in addressing it.

However, if anyone is willing to contribute a fix, the callbacks will need to be updated to accommodate the prompt formatting:

https://github.com/axolotl-ai-cloud/axolotl/blob/575e5f28ec0815654756d18be2cf6cf8a0224c20/src/axolotl/utils/callbacks/__init__.py#L577-L782
